lib/plugin/instance.rb

```diff
@@ -1018,6 +1018,31 @@ def add_topic_static_page(page, options = {}, &blk)
     StaticController::CUSTOM_PAGES[page] = blk ? { topic_id: blk } : options
   end
 
+  # Let plugin define custom unsubscribe keys,
+  # set custom instance variables on the `EmailController#unsubscribe` action,
+  # and describe what unsubscribing for that key does.
+  #
+  # The method receives a class that inherits from `Email::BaseEmailUnsubscriber`.
+  # Take a look at it to know how to implement your child class.
+  #
+  # In conjunction with this, you'll have to:
+  #
+  #  - Register a new connector under app/views/connectors/unsubscribe_options.
+  #  We'll include the HTML inside the unsubscribe form, so you can add your fields using the
+  #  instance variables you set in the controller previously. When the form it's submitted,
+  #  it sends the updated preferences to `EmailController#perform_unsubscribe`.
+  #
+  #  - Your code is responsible for creating the new UnsubscribeKey. For example,
+  #  discourse-chat creates a key with the "chat_summary" type and then use it
+  #  to add an unsubscribe link in the chat summary template.
```

Comment:
The chat implementation can change and render this comment outdated. May be best to either remove the sentence that mentions chat or, if we want to be extra clear, include some sample plugin code instead.